rfcs/000-search-revamp.md

```diff
@@ -0,0 +1,124 @@
+- analyzed matches where a user inputs keywords (e.g. "returns") and looks to find the most relevant entities
+
+Elasticsearch uses [compound queries](https://www.elastic.co/guide/en/elasticsearch/reference/current/compound-queries.html) to combine multiple subqueries together. The most common is  [Boolean](https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-bool-query.html).
+
+#### Query Optimizations
+To be effective, the union of queries discussed above still needs to be composed of high quality subqueries. 
+Here, we’ll list a handful of optimization directions and parameters we’ll experiment with:
+
+##### Fuzziness
+Allows for misspellings by specifying a maximum edit distance. See [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-fuzzy-query.html).
+
+##### Synonyms 
+This will require tuning and validation, but some users may get significant mileage from a hand-generated synonyms dictionary file. 
+We could provide a facility for specifying such a file. [Docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-synonym-graph-tokenfilter.html).
+
+##### Match Phrase Query
+In case someone searches for a phrase out of an entity description, this subquery will take word order into consideration. [Docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query-phrase.html).
+
+##### Index Fields in More Ways
+Experimenting with multiple analyzers (i.e. independently analyzing an input field via multiple analyzers) has potential benefits like normalizing stemmings and filtering stop words. 
+See [multi-field](https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html#multi-fields) about how to index a field in multiple ways, [english](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lang-analyzer.html#english-analyzer) analyzer for stemmings and stop word removal.
+Amundsen already does this to index some fields both as an unanalyzed `keyword` and as analyzed `text`.
+
+
+## Drawbacks and Alternatives
+We believe our proposal will be a significant leap in search quality that will broadly benefit all Amundsen users. Along
+with improving out-of-the-box ranking and relevance, we plan to make more parameters tunable by end users.
+
+The main drawback is around the proposed migration to ES 7, which is discussed below.
+
+## Unresolved questions
+A full ES7 migration may be difficult for some users. One option is to keep the existing ES proxy for legacy use, but
+this is complicated by needing to upgrade the Elasticsearch library in the search service to v7+. 
+Since it is recommended to use the same version of the ES lib as the server version, making v6 compatible requests
+with the v7 library may run into unexpected corner cases. 
+
+At the same time, v7 has too many new features to pass up and revamping search against the older v6 version is likely
```

Comment:
Can you list the features what we would have to pass up to stay ES6 compatible?

Reply:
Closing in favor of the comment above, so discussion on this question is in one place.